test/token/factory.t.sol

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.18;
+pragma abicoder v2;
+
+import {TrancheTokenFactory, MemberlistFactory} from "src/token/factory.sol";
+import {RestrictedToken} from "src/token/restricted.sol";
+import "forge-std/Test.sol";
+
+contract FactoryTest is Test {
+    // address(0)[0:20] + keccak("Centrifuge")[21:32]
+    bytes32 SALT = 0x000000000000000000000000000000000000000075eb27011b69f002dc094d05;
+    uint256 mainnetFork;
+    uint256 polygonFork;
+
+    function setUp() public {
+        mainnetFork = vm.createFork(vm.envString("MAINNET_RPC_URL"));
+        polygonFork = vm.createFork(vm.envString("POLYGON_RPC_URL"));
+    }
+
+    function testTokenFactoryIsDeterministicAcrossChains(
+        address sender,
+        string memory name,
+        string memory symbol,
+        uint64 poolId,
+        bytes16 trancheId,
+        uint8 decimals
+    ) public {
+        vm.selectFork(mainnetFork);
+        TrancheTokenFactory tokenFactory1 = new TrancheTokenFactory{ salt: SALT }();
+        address token1 = tokenFactory1.newTrancheToken(poolId, trancheId, name, symbol, decimals);
+
+        vm.selectFork(polygonFork);
+        TrancheTokenFactory tokenFactory2 = new TrancheTokenFactory{ salt: SALT }();
+        assertEq(address(tokenFactory1), address(tokenFactory2));
+        vm.prank(sender);
```

Comment:
Is there a reason this `vm.prank` is not higher up in the test, so it's also a different sender when deploying the factories?

Reply:
Same question and TIL on `vm.prank` :D

Reply:
Well the factories addresses will be affected by the sender, but we expect that the created token contracts shouldn't be affected by the sender address, so by putting the prank here we're also testing that anybody can deploy a token from a factory and expect the address to be the same.

Reply:
Right makes a lot of sense 👍